libsmtutil/Z3CHCInterface.cpp

```diff
@@ -82,9 +89,19 @@ pair<CheckResult, CHCSolverInterface::CexGraph> Z3CHCInterface::query(Expression
 		case z3::check_result::sat:
 		{
 			result = CheckResult::SATISFIABLE;
-			auto proof = m_solver.get_answer();
-			auto cex = cexGraph(proof);
-			return {result, cex};
+			// z3 version 4.8.8 modified Spacer to also return
+			// proofs containing nonlinear clauses.
+			if (
+				m_version.major >= 4 &&
+				m_version.minor >= 8 &&
+				m_version.build >= 8
+			)
+			{
+				auto proof = m_solver.get_answer();
+				auto cex = cexGraph(proof);
+				return {result, cex};
```

Comment:
Why not?
```suggestion
				return {result, cexGraph(proof)};
```

Or why not move these?

Also why is `cex` defined outside the `try` statement as well?

Reply:
No particular reason, can also remove it

Reply:
Updated